ARCHITECTURE.md

```diff
@@ -0,0 +1,161 @@
+focus on the elements that are specific to mix-cli.
+
+## Commands
+Since mix-cli basically mirrors the endpoints it calls under the hood,
+commands follow a common _lifecycle_. The steps of the lifcycle are exposed
+through methods that can be overridden in the implementation of individual
+commands.  When authoring new commands for mix-cli, a good place to
+start is looking at the source code of a command that offers similar
+functionality.
+
+Each command is described by a class that ultimately extends the oclif Command
+class. mix-cli command names follow the convention _topic:action_. Each
+command maps to a specific source code file under `src/commands`. For instance,
+the source code for the "projects:create" command is found in
+`src/commands/projects/create.ts`.
+
+### Command class hierarchy
+Each mix-cli command extends the MixCommand class which itself extends the
+BaseCommand class. The BaseCommand class extends the oclif Command class.
+
+```mermaid
+classDiagram
+      Command <|-- BaseCommand
+      BaseCommand <|-- MixCommand
+      MixCommand <|-- ActualCommand
+```
+
+The BaseCommand class provides methods to deal with acquiring, renewing and
+storing access tokens. It also implements the `init()` method, one of the
+lifecycle events exposed by oclif.
+
+The MixCommand class provides methods and class members that define the
+lifecycle of each mix-cli command. Follow this lifecycle to offer a
+consistent user experience when authoring new commands.
+
+## mix-cli command lifecycle
+The `run()` method of the MixCommand class implements the mix-cli command
+lifecycle. Actual mix-cli command classes extend the MixCommand class  and
+override relevant methods. It should not be necessary write a `run()` method for
+an actual mix-cli command class. The implementation of MixCommand is found in
+`src/utils/base/mix-command.ts`. MixCommand methods are listed in their order of
+use with respect to the command lifecycle.
+
+The following sections describe each part of the lifecycle and give guidance
+regarding what needs to be implemented when authoring a new command. Refer to
+the implementation of existing commands for concrete examples.
+
+### Configuration
+The first step of the mix-cli command lifecycle is to read the configuration.
+This is required for any command and is handled by default.
+
+### Command flags
+mix-cli uses named options only ("flags" in oclif parlance). The next step in
+the lifecycle is to capture the options for use across the lifecycle. Actual
+mix-cli command classes must implement the `captureOptions()` method. 
+
+#### Standard flags
+Most command flags are defined in `src/utils/flags.ts`. Import flags
+from this module when authoring a new command as it will help keep the flag
+descriptions consistent across commands. New command flags should be added to
+this module for the same reason.
+
+### Authentication
+mix.cli retrieves the access token from file storage or renews it if
```

Comment:
consistency

Reply:
Yeah, I dropped that one. The documentation sticks with the name of the package "mix-cli".